shared-helpers/src/views/sign-in/FormSignIn.tsx

```diff
@@ -34,7 +35,16 @@ const FormSignIn = ({
     window.scrollTo(0, 0)
   }
   const { LinkComponent } = useContext(NavigationContext)
-
+  const router = useRouter()
+  const listingIdRedirect = router.query?.listingId as string
+  const forgetPasswordURL =
```

Comment:
This section is a bit hard to read, a suggestion here would be defining the redirect as a single string and having one if statement for both strings that pushes the redirect onto the base.